plugins/inputs/slurm/slurm.go

```diff
@@ -0,0 +1,666 @@
+//go:generate ../../../tools/readme_config_includer/generator
+package slurm
+
+import (
+	"context"
+	_ "embed"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/plugins/common/tls"
+	"github.com/influxdata/telegraf/plugins/inputs"
+
+	goslurm "github.com/pcolladosoto/goslurm/v0038"
+)
+
+//go:embed sample.conf
+var sampleConfig string
+
+type Slurm struct {
+	URL              string          `toml:"url"`
+	Username         string          `toml:"username"`
+	Token            string          `toml:"token"`
+	IgnoredEndpoints []string        `toml:"ignored_endpoints"`
+	ResponseTimeout  config.Duration `toml:"response_timeout"`
+	tls.ClientConfig
+
+	client      *goslurm.APIClient
+	baseURL     *url.URL
+	endpointMap map[string]bool
+}
+
+func (s *Slurm) createHTTPClient(host string) *goslurm.APIClient {
+	configuration := goslurm.NewConfiguration()
+	configuration.Host = host
+	configuration.Scheme = "http"
+	configuration.UserAgent = "Telegraf Metrics Agent"
```

Comment:
We have this internal method to supply a User-Agent:
```suggestion
	configuration.UserAgent = internal.ProductToken()
```

Reply:
Thanks a ton for letting me know! I've added the change on fe4473888e72963cf25b500fb5d19bdab7a7c8c3.